app/screens/channel_add_people/channel_add_people.tsx

```diff
@@ -0,0 +1,340 @@
+            handleRemoveProfile(user.id);
+            return;
+        }
+
+        if (user.id === currentUserId) {
+            const selectedId = {
+                [currentUserId]: true,
+            };
+
+            startConversation(selectedId);
+        } else {
+            const wasSelected = selectedIds[user.id];
+            if (!wasSelected && selectedCount >= General.MAX_USERS_IN_GM) {
+                return;
+            }
+
+            const newSelectedIds = Object.assign({}, selectedIds);
+            if (!wasSelected) {
+                newSelectedIds[user.id] = user;
+            }
+
+            setSelectedIds(newSelectedIds);
+            clearSearch();
+        }
+    }, [selectedIds, currentUserId, handleRemoveProfile, startConversation, clearSearch]);
+
+    const searchUsers = useCallback(async (searchTerm: string) => {
+        const lowerCasedTerm = searchTerm.toLowerCase();
+        setLoading(true);
+
+        const results = await searchProfiles(serverUrl, lowerCasedTerm, {
+            team_id: currentTeamId,
+            allow_inactive: true,
+        });
+
+        let data: UserProfile[] = [];
+        if (results.data) {
+            data = results.data;
+        }
+
+        setSearchResults(data);
+        setLoading(false);
+    }, [restrictDirectMessage, serverUrl, currentTeamId]);
+
+    const search = useCallback(() => {
+        searchUsers(term);
+    }, [searchUsers, term]);
+
+    const onSearch = useCallback((text: string) => {
+        if (text) {
+            setTerm(text);
+            if (searchTimeoutId.current) {
+                clearTimeout(searchTimeoutId.current);
+            }
+
+            searchTimeoutId.current = setTimeout(() => {
+                searchUsers(text);
+            }, General.SEARCH_TIMEOUT_MILLISECONDS);
+        } else {
+            clearSearch();
+        }
+    }, [searchUsers, clearSearch]);
+
+    const updateNavigationButtons = useCallback(async (startEnabled: boolean) => {
+        setButtons(componentId, {
+            rightButtons: [{
+                color: theme.sidebarHeaderTextColor,
+                id: ADD_BUTTON,
+                text: formatMessage({id: 'mobile.channel_add_people.button', defaultMessage: 'Add'}),
+                showAsAction: 'always',
+                enabled: startEnabled,
+                testID: 'add_members.start.button',
+            }],
+        });
+    }, [intl.locale, theme]);
+
+    useNavButtonPressed(ADD_BUTTON, componentId, startConversation, [startConversation]);
+
+    useEffect(() => {
+        mounted.current = true;
+        updateNavigationButtons(false);
+        getProfiles();
+        return () => {
+            mounted.current = false;
+        };
+    }, []);
+
+    useEffect(() => {
+        const canStart = selectedCount > 0 && !startingConversation;
+        updateNavigationButtons(canStart);
+    }, [selectedCount > 0, startingConversation, updateNavigationButtons]);
```

Comment:
updateNavigationButtons is called twice on the first render, once in each effect. Either make the second effect a `useDidUpdate` or remove the one on the first effect.